### PR TITLE
validate extended args in config

### PR DESCRIPTION
**Pull request by deads2k**
Fixes https://github.com/openshift/origin/issues/2798.

@liggitt

### CODE REVIEW

File: pkg/cmd/server/api/validation/master.go [hunk 1]
```diff
@@ -375,3 +383,27 @@ func ValidateRoutingConfig(config api.RoutingConfig) fielderrors.ValidationError
...
+
+func ValidateAPIServerExtendedArguments(config api.ExtendedArguments) fielderrors.ValidationErrorList {
+	return ValidateExtendedArguments(config, kapp.NewAPIServer().AddFlags)
+}
+
+func ValidateControllerExtendedArguments(config api.ExtendedArguments) fielderrors.ValidationErrorList {
+	return ValidateExtendedArguments(config, cmapp.NewCMServer().AddFlags)
+}
+
+func ValidateExtendedArguments(config api.ExtendedArguments, flagFunc func(*pflag.FlagSet)) fielderrors.ValidationErrorList {
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: put shared functions in `validation.go`?

File: pkg/cmd/server/api/validation/master.go [hunk 2]
```diff
@@ -375,3 +383,27 @@ func ValidateRoutingConfig(config api.RoutingConfig) fielderrors.ValidationError
...
+	allErrs := fielderrors.ValidationErrorList{}
+
+	// check extended arguments for errors
+	for _, err := range cmdflags.Resolve(config, flagFunc) {
+		switch t := err.(type) {
+		case *fielderrors.ValidationError:
+			allErrs = append(allErrs, t)
+		default:
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("????", config, err.Error()))
```
Comment (liggitt) [inline pkg/cmd/server/api/validation/master.go]: does this include the entire config map in the message for every error of unknown type?

**Comment by liggitt** [comment]
LGTM

File: pkg/cmd/server/api/validation/master.go (hunk 2 above)
Comment (deads2k) [inline pkg/cmd/server/api/validation/master.go]: > does this include the entire config map in the message for every error of unknown type?

If we get an error type that we should never get and therefore, cannot relate to a single field, yes.

File: pkg/cmd/server/api/validation/master.go (hunk 1 above)
Comment (deads2k) [inline pkg/cmd/server/api/validation/master.go]: moved

**Comment by deads2k** [comment]
[merge]

**Pull request merged**